fly/commands/fly.go

```diff
@@ -81,4 +81,9 @@ type FlyCommand struct {
 	Completion CompletionCommand `command:"completion" description:"generate shell completion code"`
 }
 
+type TeamsParam struct {
+	AllTeams bool     `short:"a" long:"all-teams" description:"Run the command for all teams"`
+	Teams    []string `short:"n" long:"team" description:"Run the command for a list of teams"`
```

Comment:
:thinking: The short-form here clashes with `-n/--non-interactive` and doesn't share any characters with the long-form (`--team`). Any other ideas?

In a perfect world we would free up `-t/--target` for `-t/--team` instead and use some other name to identify the Concourse endpoint. Maybe `-c/--concourse`, but that's used for `-c/--config` in some places. Maybe `-e/--env`? Doesn't look like that's used anywhere.

Changing `-t/--target` at this point would break literally everyone's muscle memory, so we might want to kick that can down the road until we redesign `fly` so people get used to a whole new scheme all at once.

How about we just omit the short-form for now? Right now `fly` is really designed around having a team in the target, so until we redo the UX entirely maybe we should just have people use the long form.

Reply:
Hey @vito, thanks for the review. I take a look at the current usage of the `-n`, and it is really confusing with the normal `--non-interactive`. However, for some of the fly users they may be used to login/ set-pipeline/ see builds on team with `-n` as `--team`. I personally think it would be better to keep it for now till ux team actually have settled the idea on which short form to use in the future to reduce the amount of work if anyone is using the script on any related commands. Is that okay? 🤯
also, updated to use `TeamsFlags`

Reply:
@zoetian Hmm I think we should just omit the short-form for now. I would rather have folks explicitly spell out `--team` than specify a short flag which means something different than they might expect. We can remove the `-n` shorthand for `--team` in the other commands for consistency.

UX team isn't responsible for `fly` anymore; it's maintained as part of the Core/API track, and I think consistency is more important than anything else at least until we get around to a new flag/command scheme.